### ISSUE TITLE
Messenger Between Scope

### ISSUE THREAD

**Issue by trotsaleksandrov** [cmgmyr/laravel-messenger#209]
Hello! 
It appears for me, while trying to use default `betweenScope` located in Thread model, it doesn't work as expected. For example, `Thread::between([1,2])` will return all the Threads where users with ids `1` and `2` are participating. Am i the one who is facing this issue? Can you provide some ideas, how to manage this?
Thanks in advance!

**Comment by a-komarev** [comment]
Did you mean that it returns threads where participants are: `1, 2, 3` too?

**Comment by trotsaleksandrov** [comment]
Exactly

**Comment by a-komarev** [comment]
@trotsaleksandrov yeah... I've met such issue and just overrided a logic of getting the threads. You are free to extend Thread model with your own and write you own logic.

https://github.com/cmgmyr/laravel-messenger/blob/master/src/config/config.php#L11

**Comment by trotsaleksandrov** [comment]
That's the problem, that's it's hard to get the threads between exact users. But anyway, thanks for help :)

**Comment by a-komarev** [comment]
You can have additional column in thread there all participants will be stored as string: `1,2,3`
And when you are doing search - you can do it this string. The only thing you should to remember - always store ordered ids, because if you will save value as `1,3,2` this trick wouldn't work. And then you will need to have an index on this column.

**Comment by a-komarev** [comment]
But anyway @cmgmyr should look at this issue. As for me - current behavior is bugged.

**Comment by trotsaleksandrov** [comment]
Cannot but agree with you. It would be great if this functionality would be reviewed

**Comment by cmgmyr** [comment]
I started to look into this yesterday but wasn't able to get far before I had to get going. I am seeing the same thing on my end and agree that it should be fixed. I'll work on it.

I'm thinking that the current functionality could be useful for some. So maybe the `between` scope should return the threads only between users 1 and 2 (as intended), but adding something like `betweenLoose` that would return threads that include users 1 and 2, but also can have other participants (like current functionality). Thoughts?

**Comment by trotsaleksandrov** [comment]
Maybe it is better to split existing `between` method to two: as you mentioned, something like `betweenLoose ` and as @a-komarev mentioned `betweenOnly`. Because retrieving threads that includes users 1 and 2, might also be useful

**Comment by a-komarev** [comment]
@cmgmyr To make method `betweenLoose` was my first thought... but this will be breaking change.

**Comment by cmgmyr** [comment]
@a-komarev yup, that's exactly what I was thinking. I've also been thinking about rewriting the whole package to be a little more user friendly and not just a bunch of models.

**Comment by a-komarev** [comment]
@cmgmyr Any ETA of these plans?

**Comment by cmgmyr** [comment]
@a-komarev the `v3` plans? Unfortunately, not yet besides some planning in my head. Started a new job recently, so it's been cutting in on my free time a bit. Hopefully will get some code out soon though

**Comment by s1lviu** [comment]
For me 
`$thread = Thread::between([1, 2])->latest('updated_at')->firstOrFail();`
returns
`No query results for model [Cmgmyr\Messenger\Models\Thread].`

**Comment by AbdullahFaqeir** [comment]
Still need help with this??
